client/src/auth0-provider-with-navigate.js

```diff
@@ -0,0 +1,32 @@
+import { Auth0Provider } from "@auth0/auth0-react";
+import React from "react";
+import { useNavigate } from "react-router-dom";
+
+export const Auth0ProviderWithNavigate = ({ children }) => {
+	const navigate = useNavigate();
+	//had to expose these variables as for some reason process.env was breaking the application
+	const domain = "dev-fu5absjxigoax8c0.eu.auth0.com";
+	const clientId = "UI70osn3wvcwyWz1xLppvrgVLpiU6TCE";
+	const redirectUri = "http://localhost:3000";
```

Comment:
these were supposed to be environmental variables, for some reason when I put them in .env and called on them by process.env as described in the documentation it crashed the app with the following output `auth0-provider-with-navigate.js:12 Uncaught ReferenceError: process is not defined
    at Auth0ProviderWithNavigate (auth0-provider-with-navigate.js:12:16)
    at renderWithHooks (react-dom.development.js:16305:1)
    at mountIndeterminateComponent (react-dom.development.js:20074:1)
    at beginWork (react-dom.development.js:21587:1)
    at beginWork$1 (react-dom.development.js:27426:1)
    at performUnitOfWork (react-dom.development.js:26560:1)
    at workLoopSync (react-dom.development.js:26466:1)
    at renderRootSync (react-dom.development.js:26434:1)
    at recoverFromConcurrentError (react-dom.development.js:25850:1)
    at performConcurrentWorkOnRoot (react-dom.development.js:25750:1)
Auth0ProviderWithNavigate @ auth0-provider-with-navigate.js:12
renderWithHooks @ react-dom.development.js:16305
mountIndeterminateComponent @ react-dom.development.js:20074
beginWork @ react-dom.development.js:21587
beginWork$1 @ react-dom.development.js:27426
performUnitOfWork @ react-dom.development.js:26560
workLoopSync @ react-dom.development.js:26466
renderRootSync @ react-dom.development.js:26434
recoverFromConcurrentError @ react-dom.development.js:25850
performConcurrentWorkOnRoot @ react-dom.development.js:25750
workLoop @ scheduler.development.js:266
flushWork @ scheduler.development.js:239
performWorkUntilDeadline @ scheduler.development.js:533`

Reply:
The `process` object is a Node specific, it doesn't exist in the browser. You have to load this information into the client from server, generally by embedding it when building the webpack bundle.